core/src/state.rs

```diff
@@ -319,25 +305,25 @@ impl State {
                 new_code_hash
             );
 
-            if new_version.spec_name != old_version.spec_name {
+            if runtime_checks.name_matches && new_version.spec_name != old_version.spec_name {
                 return Err("Spec names must match.".into());
+            } else {
+                log::info!("Skipping spec name check.")
```

Comment:
This is not correct as you may have done the check. If you really want to print this, you need to create an outer if with `runtime_checks.name_matches` only as condition.